openedx/features/pakx/lms/pakx_admin_app/views.py

```diff
@@ -268,20 +271,24 @@ class CourseEnrolmentViewSet(viewsets.ModelViewSet):
     permission_classes = [CanAccessPakXAdminPanel]
 
     def enroll_users(self, request, *args, **kwargs):
-        available_courses_count = CourseOverview.objects.filter(
-            get_user_available_course_qs(self.request.user),
-            id__in=request.data["course_keys"],
-        ).count()
-        if available_courses_count != len(request.data["course_keys"]):
-            return Response(ENROLLMENT_COURSE_EXPIRED_MSG, status=status.HTTP_400_BAD_REQUEST)
-
-        user_qs = get_org_users_qs(request.user).filter(id__in=request.data["user_ids"]).values_list('id', flat=True)
         if not request.data.get("user_ids") or not request.data.get("course_keys"):
             return Response(
                 "Invalid request data! User IDs and course keys are required",
                 status=status.HTTP_400_BAD_REQUEST
             )
 
+        courses_qs = CourseOverview.objects.filter(id__in=request.data["course_keys"])
+
+        enroll_able_courses_count = courses_qs.filter(get_enroll_able_course_qs()).count()
```

Comment:
can we move some of the following functionality into utility/helper functions?
i.e helper method for each check?